components/model_analyzer/TorchBenchAnalyzer.py

```diff
@@ -182,9 +232,36 @@ def calculate_flops(self, gpu_uuid=None) -> float:
             gpu = self.gpu_factory.get_device_by_uuid(gpu_uuid)
             return gpu._sm_count * gpu._fma_count * 2 * gpu._frequency * self.gpu_metric_value[gpu_uuid][GPUFP32Active].value() / 1e+9
 
+    def calculate_gpu_peak_mem(self, gpu_uuid=None) -> float:
+        """
+        The function to calculate GPU peak memory usage for the first available GPU.
+        @return : a floating number representing GB.
+        """
+        if gpu_uuid:
+            if gpu_uuid in self.gpu_metric_value:
+                gpu = self.gpu_factory.get_device_by_uuid(gpu_uuid)
+                return self.gpu_metric_value[gpu_uuid][GPUPeakMemory].value() / 1024
+            else:
+                raise TorchBenchAnalyzerException("No available GPU with uuid ", gpu_uuid, " found!")
+        if len(self.gpu_metric_value) > 1:
+            logger.warning("There are multiple available GPUs and will only return the first one's peak memory bandwidth.")
+        gpu_uuid = next(iter(self.gpu_metric_value))
+        return self.gpu_metric_value[gpu_uuid][GPUPeakMemory].value() / 1024
+
+    def calculate_cpu_peak_mem(self, cpu_uuid=None) -> float:
```

Comment:
We need CPU peak memory to work on CPU-only machines, i.e. a machine without Nvidia or DCGM

Reply:
Yes. The TorchBenchAnalzyzer works without GPU. It checks dcgm only when we add GPU-related metrics. It includes two monitors, cpu_monitor and gpu_monitor.

Reply:
Let me double check.

Reply:
forget to move the gpu information query from analyzer initialization to start_monitor. fixed by https://github.com/pytorch/benchmark/pull/1268/commits/0aefd63b22c96f6bc9810e515d224d123cf87094